sw/device/tests/dif/dif_entropy_smoketest.c

```diff
@@ -51,14 +51,15 @@ bool test_main() {
   CHECK(dif_entropy_configure(&entropy, config) == kDifEntropyOk);
 
   uint32_t entropy_data[kEntropyDataNumWords];
+  uint32_t result = 0;
   for (uint32_t i = 0; i < kEntropyDataNumWords; ++i) {
     // wait for entropy to become available
     while (dif_entropy_read(&entropy, &entropy_data[i]) != kDifEntropyOk)
       ;
-    LOG_INFO("received %x, expectecd %x", entropy_data[i],
+    LOG_INFO("received 0x%x, expected 0x%x", entropy_data[i],
              kExpectedEntropyData[i]);
-    CHECK(entropy_data[i] == kExpectedEntropyData[i]);
+    result |= entropy_data[i] ^ kExpectedEntropyData[i];
```

Comment:
For these purposes, I think `result` can be `bool all_equal = true;`, and the xor can be ==; you can then write `CHECK(all_equal)`.